### **Context and objective:**
In this project, we'll be working with a database called The Chinook database in order to answer some business questions.
The Chinook database contains information about the artists, songs and albums from a music shop, as well as information on the shop's employees, customers and the customers purchases. This information is contained in eleven tables. 

A schema diagram is shown below to help us understand the available columns and the structure of the data.

In [1]:
from IPython.display import Image
from IPython.core.display import HTML
Image(url = "https://s3.amazonaws.com/dq-content/191/chinook-schema.svg", width = 1000, height = 1000)

The following code is used to connect Jupyter Notebook to our database 

In [1]:
%%capture
%load_ext sql
%sql sqlite:///chinook.db

'Connected: None@chinook.db'

The code below returns information on the tables and views in the database

In [2]:
%%sql
SELECT
    name,
    type
FROM sqlite_master
WHERE type IN ("table", "view");

Done.


[('album', 'table'),
 ('artist', 'table'),
 ('customer', 'table'),
 ('employee', 'table'),
 ('genre', 'table'),
 ('invoice', 'table'),
 ('invoice_line', 'table'),
 ('media_type', 'table'),
 ('playlist', 'table'),
 ('playlist_track', 'table'),
 ('track', 'table'),
 ('main_table', 'view'),
 ('m_table', 'view'),
 ('m_tab', 'view')]

Write some queries to display some tables of the database

In [3]:
%%sql
SELECT * FROM employee;

Done.


[(1, 'Adams', 'Andrew', 'General Manager', None, '1962-02-18 00:00:00', '2016-08-14 00:00:00', '11120 Jasper Ave NW', 'Edmonton', 'AB', 'Canada', 'T5K 2N1', '+1 (780) 428-9482', '+1 (780) 428-3457', 'andrew@chinookcorp.com'),
 (2, 'Edwards', 'Nancy', 'Sales Manager', 1, '1958-12-08 00:00:00', '2016-05-01 00:00:00', '825 8 Ave SW', 'Calgary', 'AB', 'Canada', 'T2P 2T3', '+1 (403) 262-3443', '+1 (403) 262-3322', 'nancy@chinookcorp.com'),
 (3, 'Peacock', 'Jane', 'Sales Support Agent', 2, '1973-08-29 00:00:00', '2017-04-01 00:00:00', '1111 6 Ave SW', 'Calgary', 'AB', 'Canada', 'T2P 5M5', '+1 (403) 262-3443', '+1 (403) 262-6712', 'jane@chinookcorp.com'),
 (4, 'Park', 'Margaret', 'Sales Support Agent', 2, '1947-09-19 00:00:00', '2017-05-03 00:00:00', '683 10 Street SW', 'Calgary', 'AB', 'Canada', 'T2P 5G3', '+1 (403) 263-4423', '+1 (403) 263-4289', 'margaret@chinookcorp.com'),
 (5, 'Johnson', 'Steve', 'Sales Support Agent', 2, '1965-03-03 00:00:00', '2017-10-17 00:00:00', '7727B 41 Ave', 'Calgary', 'AB', 'Canada', 'T3B 1Y7', '1 (780) 836-9987', '1 (780) 836-9543', 'steve@chinookcorp.com'),
 (6, 'Mitchell', 'Michael', 'IT Manager', 1, '1973-07-01 00:00:00', '2016-10-17 00:00:00', '5827 Bowness Road NW', 'Calgary', 'AB', 'Canada', 'T3B 0C5', '+1 (403) 246-9887', '+1 (403) 246-9899', 'michael@chinookcorp.com'),
 (7, 'King', 'Robert', 'IT Staff', 6, '1970-05-29 00:00:00', '2017-01-02 00:00:00', '590 Columbia Boulevard West', 'Lethbridge', 'AB', 'Canada', 'T1K 5N8', '+1 (403) 456-9986', '+1 (403) 456-8485', 'robert@chinookcorp.com'),
 (8, 'Callahan', 'Laura', 'IT Staff', 6, '1968-01-09 00:00:00', '2017-03-04 00:00:00', '923 7 ST NW', 'Lethbridge', 'AB', 'Canada', 'T1H 1Y8', '+1 (403) 467-3351', '+1 (403) 467-8772', 'laura@chinookcorp.com')]

In [4]:
%%sql
SELECT * FROM album
   LIMIT 10;

Done.


[(1, 'For Those About To Rock We Salute You', 1),
 (2, 'Balls to the Wall', 2),
 (3, 'Restless and Wild', 2),
 (4, 'Let There Be Rock', 1),
 (5, 'Big Ones', 3),
 (6, 'Jagged Little Pill', 4),
 (7, 'Facelift', 5),
 (8, 'Warner 25 Anos', 6),
 (9, 'Plays Metallica By Four Cellos', 7),
 (10, 'Audioslave', 8)]

In [5]:
%%sql
SELECT * FROM track
   LIMIT 10;

Done.


[(1, 'For Those About To Rock (We Salute You)', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 343719, 11170334, 0.99),
 (2, 'Balls to the Wall', 2, 2, 1, None, 342562, 5510424, 0.99),
 (3, 'Fast As a Shark', 3, 2, 1, 'F. Baltes, S. Kaufman, U. Dirkscneider & W. Hoffman', 230619, 3990994, 0.99),
 (4, 'Restless and Wild', 3, 2, 1, 'F. Baltes, R.A. Smith-Diesel, S. Kaufman, U. Dirkscneider & W. Hoffman', 252051, 4331779, 0.99),
 (5, 'Princess of the Dawn', 3, 2, 1, 'Deaffy & R.A. Smith-Diesel', 375418, 6290521, 0.99),
 (6, 'Put The Finger On You', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 205662, 6713451, 0.99),
 (7, "Let's Get It Up", 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 233926, 7636561, 0.99),
 (8, 'Inject The Venom', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 210834, 6852860, 0.99),
 (9, 'Snowballed', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 203102, 6599424, 0.99),
 (10, 'Evil Walks', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 263497, 8611245, 0.99)]

In [6]:
%%sql
SELECT * FROM media_type
   LIMIT 10;

Done.


[(1, 'MPEG audio file'),
 (2, 'Protected AAC audio file'),
 (3, 'Protected MPEG-4 video file'),
 (4, 'Purchased AAC audio file'),
 (5, 'AAC audio file')]

#### Next step:
We are tasked of selecting the first three albums from a list of four after the Chinook store signed a new deal with a record label. All four artists don't have any tracks in the store right now. We have the artist names and the genre of music they produce:

|   **Artist Name**    |   **Genre**    |
|----------------------|----------------|
|Regal                 |Hip-Hop         |
|Red Tone              |Punk            |
|Meteor and the Girls  |Pop             |
|Slim Jim Bites        |Blues           |

We aim to advertise the new albums in the USA, so we need to find out which genres sell the best in the USA. 
First we'll need to write a query to find out which genres sell the most tracks and then make a recommendation on which albums we should purchase in the store based on our findings.

In [7]:
%%sql
WITH track_genre AS
    (SELECT 
         track.track_id AS track_id,
         track.name AS track_name,
         track.composer AS composer,
         genre.name AS genre_name,
         customer.country AS country
     FROM customer
     INNER JOIN invoice ON invoice.customer_id = customer.customer_id
     INNER JOIN invoice_line ON invoice_line.invoice_id = invoice.invoice_id
     INNER JOIN track ON track.track_id = invoice_line.track_id
     INNER JOIN genre ON genre.genre_id = track.genre_id
     WHERE country = "USA"
    )
    
SELECT 
    genre_name,
    COUNT(track_genre.track_id) AS number_of_tracks,
    ROUND((CAST(COUNT(track_genre.track_id) AS FLOAT) / (SELECT COUNT(*) FROM track_genre)) * 100, 2) AS percentage,
    country
    FROM track_genre
    WHERE genre_name LIKE "%Hip%Hop%"
       OR genre_name LIKE "%Punk%"
       OR genre_name LIKE "%Pop%"
       OR genre_name LIKE "%Blues%"
    GROUP BY genre_name
    ORDER BY number_of_tracks DESC;

Done.


[('Alternative & Punk', 130, 12.37, 'USA'),
 ('Blues', 36, 3.43, 'USA'),
 ('Pop', 22, 2.09, 'USA'),
 ('Hip Hop/Rap', 20, 1.9, 'USA')]

Based on our findings, we can conclude that the genres that sell the most in the USA are the following:

1. Punk (Artist name: Red Tone)
2. Blues (Artist name: Slim Jim Bites)
3. Pop (Artist name: Meteor and the Girls)

So our recommendation is to purchase tracks from those three genres.

#### **Sales Support Agents' Performance**
Each customer in the Chinook store gets assigned to a sales support agent when they first make a purchase. We are asked to analyze the purchases of each customer belonging to an employee in order to get some insights on each sales support agent performance.  
To do that, we'll write a query that finds the total dollar amount of sales attributed to each employee.

In [8]:
%%sql
SELECT
    employee.first_name ||" "|| employee.last_name AS employee_name,
    employee.hire_date AS employee_hire_date,
    employee.country AS employee_country,
    customer.company As customer_company,
    customer.country As customer_country,
    SUM(invoice.total) AS total_amount
FROM employee
INNER JOIN customer ON customer.support_rep_id = employee.employee_id
INNER JOIN invoice ON invoice.customer_id = customer.customer_id
GROUP BY employee_name;

Done.


[('Jane Peacock', '2017-04-01 00:00:00', 'Canada', None, 'United Kingdom', 1731.5100000000039),
 ('Margaret Park', '2017-05-03 00:00:00', 'Canada', None, 'USA', 1584.0000000000034),
 ('Steve Johnson', '2017-10-17 00:00:00', 'Canada', 'Telus', 'Canada', 1393.920000000002)]

#### **Insights:**
- There are three sales support agents. All of them are from Canada.
- All three employees were hired in 2017. Jane Peacock was hired first in april 2017. Margret Park was hired a month later and Steve Johnson was hired five months after Margaret Park. We can see from our results that the total amount assigned to the three employees are greater when they are hired before their collegues at the company. 
- The difference in performance between the employees is related to their hiring dates.

#### Analyze the sales data for customers from each different country:

In this step, we have been directed to calculate data for each country, on the:
- total number of customers
- total value of sales
- average value of sales per customer
- average order value

There are some countries with one customer, we'll group these customers as "Other" and force its order to last in our analysis.
The results will be sorted from highest to lowest, with the "Other" column at the bottom.

In [9]:
%%sql
WITH other_country AS
   (SELECT 
         CASE
             WHEN (SELECT 
                        COUNT(*) 
                   FROM customer
                   WHERE country = c.country
                  ) = 1 THEN "Other"
             ELSE c.country
         END AS countries,
         c.customer_id,
         il.*
    FROM invoice_line il
    INNER JOIN invoice i ON i.invoice_id = il.invoice_id
    INNER JOIN customer c ON c.customer_id = i.customer_id
   ),

    purchase_country AS
    (SELECT
        other_country.countries AS country,
        COUNT(DISTINCT other_country.customer_id) AS number_customers,
        SUM(other_country.unit_price) AS total_sales,
        SUM(other_country.unit_price) / COUNT(DISTINCT other_country.customer_id) AS avg_customer_sales,
        SUM(other_country.unit_price) / COUNT(DISTINCT other_country.invoice_id) AS avg_order_val
    FROM other_country
         GROUP BY country
         ORDER BY total_sales DESC
    )
    
SELECT
    country,
    number_customers,
    total_sales,
    avg_customer_sales,
    avg_order_val
FROM
    (
     SELECT
        purchase_country.*,
        CASE
           WHEN country = "Other" THEN 1
           ELSE 0
        END AS sort
     FROM purchase_country
    )
ORDER BY sort ASC;

Done.


[('USA', 13, 1040.490000000008, 80.03769230769292, 7.942671755725252),
 ('Canada', 8, 535.5900000000034, 66.94875000000043, 7.047236842105309),
 ('Brazil', 5, 427.68000000000245, 85.53600000000048, 7.011147540983647),
 ('France', 5, 389.0700000000021, 77.81400000000042, 7.781400000000042),
 ('Germany', 4, 334.6200000000016, 83.6550000000004, 8.161463414634186),
 ('Czech Republic', 2, 273.24000000000103, 136.62000000000052, 9.108000000000034),
 ('United Kingdom', 3, 245.52000000000078, 81.84000000000026, 8.768571428571457),
 ('Portugal', 2, 185.13000000000022, 92.56500000000011, 6.3837931034482835),
 ('India', 2, 183.1500000000002, 91.5750000000001, 8.72142857142858),
 ('Other', 15, 1094.9400000000085, 72.99600000000056, 7.448571428571486)]

#### **Invoice categorization:**
The Chinook store is setup in a way that allows customers to purchase in one of the two ways:

- purchase a whole album
- purchase a collection of one or more individual tracks

The store does not let customers to purchase a whole album, and then add individual tracks to the same purchase.
Management is considering changing their strategy to save money. The new considered strategy is to purchase only the most popular tracks from an album from a record company instead of purchasing the whole album.

We have been asked to make an analysis to find the percentage of purchases of whole albums vs individual tracks. This data will be used to understand the impact this decision might have on overall revenue.

To achieve our objective, we need to write a query that categorizes each invoice as either an album purchase or not and calculates the following summary statistics:

- Number of invoices
- Percentage of invoices

In [10]:
%%sql
SELECT 
     * 
FROM invoice_line
INNER JOIN track ON track.track_id = invoice_line.track_id;

Done.


[(1, 1, 1158, 0.99, 1, 1158, 'Right Next Door to Hell', 91, 2, 1, None, 182321, 3175950, 0.99),
 (2, 1, 1159, 0.99, 1, 1159, "Dust N' Bones", 91, 2, 1, None, 298374, 5053742, 0.99),
 (3, 1, 1160, 0.99, 1, 1160, 'Live and Let Die', 91, 2, 1, None, 184016, 3203390, 0.99),
 (4, 1, 1161, 0.99, 1, 1161, "Don't Cry (Original)", 91, 2, 1, None, 284744, 4833259, 0.99),
 (5, 1, 1162, 0.99, 1, 1162, 'Perfect Crime', 91, 2, 1, None, 143637, 2550030, 0.99),
 (6, 1, 1163, 0.99, 1, 1163, "You Ain't the First", 91, 2, 1, None, 156268, 2754414, 0.99),
 (7, 1, 1164, 0.99, 1, 1164, 'Bad Obsession', 91, 2, 1, None, 328282, 5537678, 0.99),
 (8, 1, 1165, 0.99, 1, 1165, 'Back off Bitch', 91, 2, 1, None, 303436, 5135662, 0.99),
 (9, 1, 1166, 0.99, 1, 1166, "Double Talkin' Jive", 91, 2, 1, None, 203637, 3520862, 0.99),
 (10, 1, 1167, 0.99, 1, 1167, 'November Rain', 91, 2, 1, None, 537540, 8923566, 0.99),
 (11, 1, 1168, 0.99, 1, 1168, 'The Garden', 91, 2, 1, None, 322175, 5438862, 0.99),
 (12, 1, 1169, 0.99, 1, 1169, 'Garden of Eden', 91, 2, 1, None, 161539, 2839694, 0.99),
 (13, 1, 1170, 0.99, 1, 1170, "Don't Damn Me", 91, 2, 1, None, 318901, 5385886, 0.99),
 (14, 1, 1171, 0.99, 1, 1171, 'Bad Apples', 91, 2, 1, None, 268351, 4567966, 0.99),
 (15, 1, 1172, 0.99, 1, 1172, 'Dead Horse', 91, 2, 1, None, 257600, 4394014, 0.99),
 (16, 1, 1173, 0.99, 1, 1173, 'Coma', 91, 2, 1, None, 616511, 10201342, 0.99),
 (17, 2, 3476, 0.99, 1, 3476, 'Help Yourself', 322, 2, 9, 'Freddy James, Jimmy hogarth & Larry Stock', 300884, 5029266, 0.99),
 (18, 2, 482, 0.99, 1, 482, 'Good Riddance (Time Of Your Life)', 39, 1, 4, 'Billie Joe Armstrong -Words Green Day -Music', 153600, 5075241, 0.99),
 (19, 2, 2701, 0.99, 1, 2701, 'Blinded By Rainbows', 218, 1, 1, 'Jagger/Richards', 273946, 8971343, 0.99),
 (20, 2, 1641, 0.99, 1, 1641, 'Gallows Pole', 134, 1, 1, 'Traditional', 296228, 9757151, 0.99),
 (21, 2, 1119, 0.99, 1, 1119, 'Toda Menina Baiana', 73, 1, 7, 'Gilberto Gil', 278177, 9351000, 0.99),
 (22, 2, 2324, 0.99, 1, 2324, 'Pretty Persuasion', 190, 1, 4, 'R.E.M.', 229929, 7577754, 0.99),
 (23, 2, 201, 0.99, 1, 201, 'Keep It To Myself (Aka Keep It To Yourself)', 20, 1, 6, 'Sonny Boy Williamson [i]', 166060, 5487056, 0.99),
 (24, 2, 819, 0.99, 1, 819, 'Talk About Love', 66, 1, 1, 'roger glover', 247823, 8072171, 0.99),
 (25, 2, 392, 0.99, 1, 392, 'Tim Tim Por Tim Tim', 34, 1, 7, 'Vários', 213237, 7143328, 0.99),
 (26, 2, 2651, 0.99, 1, 2651, "Can't Stand Losing You", 215, 1, 1, 'G M Sumner', 181159, 5971983, 0.99),
 (27, 3, 2516, 0.99, 1, 2516, 'Black Hole Sun', 203, 1, 1, 'Soundgarden', 320365, 10425229, 0.99),
 (28, 3, 2646, 0.99, 1, 2646, 'I Looked At You', 214, 1, 1, 'Robby Krieger, Ray Manzarek, John Densmore, Jim Morrison', 142080, 4663988, 0.99),
 (29, 4, 3448, 0.99, 1, 3448, 'Lamentations of Jeremiah, First Set \\ Incipit Lamentatio', 314, 2, 24, 'Thomas Tallis', 69194, 1208080, 0.99),
 (30, 4, 2560, 0.99, 1, 2560, 'Violent Pornography', 207, 1, 3, 'Dolmayan, John/Malakian, Daron/Odadjian, Shavo', 211435, 6985960, 0.99),
 (31, 4, 3336, 0.99, 1, 3336, 'War Pigs', 260, 4, 23, None, 234013, 8052374, 0.99),
 (32, 4, 829, 0.99, 1, 829, "Let's Get Rocked", 67, 1, 1, None, 296019, 9724150, 0.99),
 (33, 4, 1872, 0.99, 1, 1872, 'Attitude', 153, 1, 3, 'Hetfield, Ulrich', 315898, 10335734, 0.99),
 (34, 4, 748, 0.99, 1, 748, 'Dealer', 58, 1, 1, 'Bolin/Coverdale', 230922, 7591066, 0.99),
 (35, 4, 1778, 0.99, 1, 1778, "You're What's Happening (In The World Today)", 146, 1, 14, 'Allen Story/George Gordy/Robert Gordy', 142027, 4631104, 0.99),
 (36, 4, 2514, 0.99, 1, 2514, 'Spoonman', 203, 1, 1, 'Chris Cornell', 248476, 8289906, 0.99),
 (37, 5, 1986, 0.99, 1, 1986, 'Intro', 163, 1, 1, 'Kurt Cobain', 52218, 1688527, 0.99),
 (38, 5, 1987, 0.99, 1, 1987, 'School', 163, 1, 1, 'Kurt Cobain', 160235, 5234885, 0.99),
 (39, 5, 1988, 0.99, 1, 1988, 'Drain You', 163, 1, 1, 'Kurt Cobain', 215196, 7013175, 0.99),
 (40, 5, 1989, 0.99, 1, 1989, 'Aneurysm', 163, 1, 1, 'Nirvana', 271516, 8862545, 0.99)

In [11]:
%%sql
WITH category AS 
             (SELECT 
                   invoice_line.invoice_id,
                   CASE
                      WHEN 
                         COUNT(DISTINCT(track.album_id)) = 1
                         AND
                         COUNT(invoice_line.track_id) = c.track_count
                       THEN "album"
                       ELSE "individual_tracks"
                   END AS purchase_category 
              FROM invoice_line
              INNER JOIN track ON track.track_id = invoice_line.track_id
              INNER JOIN (SELECT COUNT(*) AS track_count, album_id
                          FROM track
                          GROUP BY track.album_id) c ON c.album_id = track.album_id
              GROUP BY invoice_line.invoice_id
             )
SELECT 
    purchase_category,
    COUNT(*) AS number_of_invoices,
    ROUND(CAST(COUNT(*) AS FLOAT) / (SELECT COUNT(*) FROM category), 2) AS percentage_invoices
    FROM category
    GROUP BY purchase_category;

Done.


[('album', 114, 0.19), ('individual_tracks', 500, 0.81)]

The same result can be obtained with a second coding method using EXCEPT operator.

In [12]:
%%sql
WITH category_1 AS
                 (SELECT 
                      *
                  FROM invoice_line
                INNER JOIN track ON track.track_id = invoice_line.track_id
                INNER JOIN (SELECt * FROM track
                            GROUP BY track.album_id) c ON c.album_id = track.album_id
                GROUP BY invoice_line.invoice_id
                 )
SELECT * FROM category_1;

Done.


[(16, 1, 1173, 0.99, 1, 1173, 'Coma', 91, 2, 1, None, 616511, 10201342, 0.99, 1173, 'Coma', 91, 2, 1, None, 616511, 10201342, 0.99),
 (17, 2, 3476, 0.99, 1, 3476, 'Help Yourself', 322, 2, 9, 'Freddy James, Jimmy hogarth & Larry Stock', 300884, 5029266, 0.99, 3477, 'Amy Amy Amy (Outro)', 322, 2, 9, 'Astor Campbell, Delroy "Chris" Cooper, Donovan Jackson, Dorothy Fields, Earl Chinna Smith, Felix Howard, Gordon Williams, James Moody, Jimmy McHugh, Matt Rowe, Salaam Remi & Stefan Skarbek', 663426, 10564704, 0.99),
 (28, 3, 2646, 0.99, 1, 2646, 'I Looked At You', 214, 1, 1, 'Robby Krieger, Ray Manzarek, John Densmore, Jim Morrison', 142080, 4663988, 0.99, 2649, 'The End', 214, 1, 1, 'Robby Krieger, Ray Manzarek, John Densmore, Jim Morrison', 701831, 22927336, 0.99),
 (29, 4, 3448, 0.99, 1, 3448, 'Lamentations of Jeremiah, First Set \\ Incipit Lamentatio', 314, 2, 24, 'Thomas Tallis', 69194, 1208080, 0.99, 3492, 'Sing Joyfully', 314, 2, 24, 'William Byrd', 133768, 2256484, 0.99),
 (53, 5, 2002, 0.99, 1, 2002, 'Blew', 163, 1, 1, 'Kurt Cobain', 216346, 7096936, 0.99, 2002, 'Blew', 163, 1, 1, 'Kurt Cobain', 216346, 7096936, 0.99),
 (54, 6, 1045, 0.99, 1, 1045, 'Fly Me To The Moon', 83, 1, 12, 'bart howard', 149263, 4856954, 0.99, 1056, 'L.A. Is My Lady', 83, 1, 12, 'alan bergman/marilyn bergman/peggy lipton jones/quincy jones', 193175, 6378511, 0.99),
 (56, 7, 3477, 0.99, 1, 3477, 'Amy Amy Amy (Outro)', 322, 2, 9, 'Astor Campbell, Delroy "Chris" Cooper, Donovan Jackson, Dorothy Fields, Earl Chinna Smith, Felix Howard, Gordon Williams, James Moody, Jimmy McHugh, Matt Rowe, Salaam Remi & Stefan Skarbek', 663426, 10564704, 0.99, 3477, 'Amy Amy Amy (Outro)', 322, 2, 9, 'Astor Campbell, Delroy "Chris" Cooper, Donovan Jackson, Dorothy Fields, Earl Chinna Smith, Felix Howard, Gordon Williams, James Moody, Jimmy McHugh, Matt Rowe, Salaam Remi & Stefan Skarbek', 663426, 10564704, 0.99),
 (72, 8, 2946, 0.99, 1, 2946, 'When I Look At The World', 233, 1, 1, 'Adam Clayton, Bono, Larry Mullen, The Edge', 257776, 8500491, 0.99, 2948, 'Grace', 233, 1, 1, 'Adam Clayton, Bono, Larry Mullen, The Edge', 330657, 10877148, 0.99),
 (78, 9, 3456, 0.99, 1, 3456, "You Know I'm No Good", 321, 2, 14, None, 256946, 4133694, 0.99, 3466, 'Rehab (Hot Chip Remix)', 321, 2, 14, None, 418293, 6670600, 0.99),
 (87, 10, 3290, 0.99, 1, 3290, 'The Zoo', 257, 2, 1, None, 332740, 5550779, 0.99, 3299, 'Send Me an Angel', 257, 2, 1, None, 273041, 4581492, 0.99),
 (98, 11, 3452, 0.99, 1, 3452, 'SCRIABIN: Prelude in B Major, Op. 11, No. 11', 318, 4, 24, None, 101293, 3819535, 0.99, 3452, 'SCRIABIN: Prelude in B Major, Op. 11, No. 11', 318, 4, 24, None, 101293, 3819535, 0.99),
 (101, 12, 2663, 0.99, 1, 2663, 'Message in a Bottle (new classic rock mix)', 215, 1, 1, 'G M Sumner', 290951, 9640349, 0.99, 2663, 'Message in a Bottle (new classic rock mix)', 215, 1, 1, 'G M Sumner', 290951, 9640349, 0.99),
 (103, 13, 2553, 0.99, 1, 2553, 'Where The River Goes', 206, 1, 1, 'D. DeLeo/Kretz/Weiland', 505991, 16468904, 0.99, 2553, 'Where The River Goes', 206, 1, 1, 'D. DeLeo/Kretz/Weiland', 505991, 16468904, 0.99),
 (104, 14, 541, 0.99, 1, 541, 'Banho De Lua', 42, 1, 4, 'B. de Filippi - F. Migliaci - Versão: Fred Jorge', 221831, 7232123, 0.99, 542, 'Meu Refrigerador Não Funciona', 42, 1, 4, 'Arnaldo Baptista - Rita Lee - Sérgio Dias', 382981, 12495906, 0.99),
 (108, 15, 3336, 0.99, 1, 3336, 'War Pigs', 260, 4, 23, None, 234013, 8052374, 0.99, 3336, 'War Pigs', 260, 4, 23, None, 234013, 8052374, 0.99),
 (114, 16, 3460, 0.99, 1, 3460, 'Love Is a Losing Game', 321, 2, 14, None, 154386, 2509409, 0.99, 3466, 'Rehab (Hot Chip Remix)', 321, 2, 14, None, 418293, 6670600, 0.99),
 (125, 17, 2564, 0.99, 1, 2564, 'Lost in Hollywood', 207, 1, 3, 'Tankian, Serj', 320783, 10535158, 0.99, 2564, 'Lost in Hollywood', 207, 1, 3, 'Tankian, Serj', 320783, 10535158, 0.99),
 (133, 18, 2736, 0.99, 1, 2736, 'Boris The Spider', 221, 1, 1, 'John Entwistle', 149472, 4835202, 0.99, 2750, 'You Better You Bet', 221, 1, 1,

Album puchases account for 19% of total purchases while purchases of individual tracks account for 81%. Based on these findings, we would logically recommend the store to purchase a whole album instead of purchasing individual tracks because we may lose about 20% of the store's revenue.